integration/cri-o/crio.bats

```diff
@@ -16,24 +16,21 @@
 
 load helpers
 
-function setup() {
-	start_crio
-}
-
 function teardown() {
-	stop_crio
-	sudo umount "$TESTDIR/crio/overlay"
 	cleanup_test
 }
 
 @test "ctr not found correct error message" {
-	run crictl inspect randomid
+	start_crio
```

Comment:
Why do all the tests now call what was happening in the per-test `setup()` and `teardown()` functions? This seems like a retrograde step and isn't explained in the commit.

Reply:
@jodh-intel those changes are following the bats from version 1.9 of CRIO. And those tests start and stop crio for every test. That's why this PR is also making sure that we start the crio service only when needed. Otherwise we end up with some failures when bats are trying to stop the service. Indeed, they kill the crio process, but it gets restarted automatically.

Reply:
I'm still confused. Previously, `crio.bats` had a `setup()` and `teardown()` and those started and stopped the crio service respectively.

But now, you've explicitly added a start and stop to every test - why can't we continue to use `setup()` and `teardown()` which do the same, but in a simpler way?

Oh... I think I've just worked out what you're doing now. "a simple copy/paste" has only just registered ;)

The `crio.bats` you're adding is a copy of https://github.com/kubernetes-incubator/cri-o/blob/master/test/ctr.bats right?

If so, why? Why can't we just `go get -d` / `git clone` (or even `curl`) the version we want to test with?

Reply:
Another observation: the `crio.bats` on this PR is very similar to https://github.com/kubernetes-incubator/cri-o/blob/master/test/ctr.bats, but we've added our own copyright header that isn't in the original...?

Reply:
Yes that's exactly what you said, I'm doing a copy/paste here from the CRI-O repo. The reason I don't do a simple `go get ...` to retrieve the bats file, is that we don't run all the bats, because they don't pass all. That's why we have to pick only the one passing for us, and that's also the reason why we have our own copyright on this one.

Reply:
I am fine with putting CRI-O copyright here, if that's what needs to be done. @grahamwhaley WDYT ?

Reply:
Oh what fun - the original does not have an explicit copyright in it, and I don't see an obvious `COPYRIGHT` file in the project/repo. I think we should attribute the file (say at the top basically 'this is a modified copy of the CRI-O file from https://github.com/kubernetes-incubator/cri-o/blob/master/test/ctr.bats). Don't make it an actual copyright notice (it's not really for us to say that - thats up to the CRI-O folks) - but make it clear where we cloned it from eh?

Reply:
@sboeuf - I'm still unclear why we need to have this stored in our repository? I understand we can't use `go get`, but surely we could have the test scripts do something like the following?:

```
commit="6de90e046a0009b69dcfbc53335d8742532004d8"
curl -O "https://raw.githubusercontent.com/kubernetes-incubator/cri-o/${commit}/test/ctr.bats"
```

That way, we don't need to store the file and we don't need to worry about copyrights surely?

Reply:
I think it is because as @sboeuf said, we modify the script to remove some of the tests that we don't run? so, unless we could automate that with some sed/awk magic or similar, maybe having a locally modified copy is the easiest route at present?

Reply:
Well we thought it would be easier to get our own test file, that we copy when running the CI. I am not modifying any behavior here, just following what we've been doing for months now. You can try to move to something different as you described here, but this needs to be done in a different PR, I just want to make sure here that we run the tests based on the latest scripts from CRI-O.

Reply:
@sboeuf - yes, I see that.

@chavafg - we might want to find a better way of dealing with this. If copying is the best process, we need to document it somewhere in a `README.md` imho (on a separate PR).

Reply:
sure, I opened https://github.com/clearcontainers/tests/issues/921 to address this concern.